bigtable/internal/testproxy/proxy.go

```diff
@@ -789,7 +789,58 @@ func (s *goTestProxyServer) BulkMutateRows(ctx context.Context, req *pb.MutateRo
 // CheckAndMutateRow responds to the CheckAndMutateRow RPC. This method applies
 // one mutation if a condition is true and another mutation if it is false.
 func (s *goTestProxyServer) CheckAndMutateRow(ctx context.Context, req *pb.CheckAndMutateRowRequest) (*pb.CheckAndMutateRowResult, error) {
-	return nil, stat.Error(codes.Unimplemented, "CheckAndMutateRow not implemented")
+	s.clientsLock.Lock()
+	btc, exists := s.clientIDs[req.ClientId]
+	s.clientsLock.Unlock()
+
+	if !exists {
+		log.Printf("received invalid ClientID: %s\n", req.ClientId)
+		return nil, stat.Error(codes.InvalidArgument,
+			fmt.Sprintf("%s: ClientID does not exist", logLabel))
+	}
+
+	rrq := req.GetRequest()
+	if rrq == nil {
+		log.Printf("request to CheckAndMutateRow is missing inner request: received: %v", req)
+		return nil, stat.Error(codes.InvalidArgument, "request to CheckAndMutateRow() is missing inner request")
+	}
+
+	trueMuts := mutationFromProto(rrq.TrueMutations)
+	falseMuts := mutationFromProto(rrq.FalseMutations)
+
+	rfPb := rrq.PredicateFilter
+	f := bigtable.PassAllFilter()
+
+	if rfPb != nil {
+		f = *filterFromProto(rfPb)
+	}
+
+	c := bigtable.NewCondMutation(f, trueMuts, falseMuts)
+
+	var matched bool
+	ao := bigtable.GetCondMutationResult(&matched)
```

Comment:
nit: let's keep declarations closest to their use. This `ao` and matched could move down to line 823

Reply:
Done.